### PR TITLE
Support custom HTTP methods for HTTP/2 

### PR DESCRIPTION
**Pull request by johanandren**
References #2394

### CODE REVIEW

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 1]
```diff
@@ -17,9 +17,14 @@ import akka.http.impl.engine.server.HttpAttributes
 import akka.http.impl.engine.ws.ByteStringSinkProbe
 import akka.http.impl.util.AkkaSpecWithMaterializer
 import akka.http.impl.util.LogByteStringTools
+import akka.http.javadsl.Http
```
Comment (raboof) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:20]: Is this used?

**Review by raboof — APPROVED** [review]
banana.

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala (hunk 1 above)
Comment (johanandren) [reply akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala:20]: No, probably Intellij, I should probably turn off that autoimport feature it messes with my PRs all the time...

**Review by jrudolph — APPROVED** [review]
:banana: /lgtm HTTP/2.0

The ticket had another part to it which was providing the right error message / response when parsing fails.

**Pull request merged**